sql/core/src/test/scala/org/apache/spark/sql/sources/PartitionedWriteSuite.scala

```diff
@@ -164,4 +165,48 @@ class PartitionedWriteSuite extends QueryTest with SharedSparkSession {
       assert(e.getMessage.contains("Found duplicate column(s) b, b: `b`;"))
     }
   }
+
+  test("SPARK-27194 SPARK-29302: Fix commit collision in dynamic partition overwrite mode") {
```

Comment:
I tried the test and it can pass without this fix. Can you take a look?

Reply:
@WinkerDu update this case?

Reply:
@Ngone51 already fix it, please have an another try :)